bindings/java/src/error.rs

```diff
@@ -115,7 +124,13 @@ impl Debug for Error {
 
 impl Display for Error {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        Display::fmt(&self.inner, f)
+        Display::fmt(&self.inner, f)?;
+        if self.backtrace.status() == BacktraceStatus::Captured {
```

Comment:
`opendal::Error` implements `Display` and `Debug` very carefully, how about doing the same thing for java binding's error?

Reply:
This is not the major issue here. I implemented `Display` and `Debug` for Java binding's Error also, but I'd prefer to use Display with backtrace. Most of what I need is Display's result, only the backtrace is necessary for this binding debugging requirement.

You can pull the code to check the full impl.

Reply:
Seems using debug meets what I need here. I'll change the impl now.

Reply:
Out of curiousity, when does `if f.alternate()` hit?